### PR TITLE
Remove explicit ComponentCatalog parameter

### PR DESCRIPTION
**Pull request by beneyal**
ValidateNodes and EntryPointNode now use the ComponentCatalog
property of IHostEnvironment.

Following the conversation with @Ivanidzo4ka at #1135

### CODE REVIEW

File: src/Microsoft.ML.Data/EntryPoints/EntryPointNode.cs [hunk 1]
```diff
@@ -561,39 +559,36 @@ private void SetColumnArgument(IChannel ch, object inputInstance, string argName
...
             env.CheckValue(inputBindingMap, nameof(inputBindingMap));
             env.CheckValue(inputMap, nameof(inputMap));
             env.CheckValue(outputMap, nameof(outputMap));
             ComponentCatalog.EntryPointInfo info;
-            bool success = catalog.TryFindEntryPoint(entryPointName, out info);
+            bool success = env.ComponentCatalog.TryFindEntryPoint(entryPointName, out info);
             env.Assert(success);
 
-            var inputBuilder = new InputBuilder(env, info.InputType, catalog);
+            var inputBuilder = new InputBuilder(env, info.InputType, env.ComponentCatalog);
```
Comment (eerhardt) [inline src/Microsoft.ML.Data/EntryPoints/EntryPointNode.cs]: Can you remove the ComponentCatalog param from this constructor too?

File: src/Microsoft.ML.Data/EntryPoints/EntryPointNode.cs [hunk 2]
```diff
@@ -561,39 +559,36 @@ private void SetColumnArgument(IChannel ch, object inputInstance, string argName
...
             var outputHelper = new OutputHelper(env, info.OutputType);
 
             using (var ch = env.Start("Create EntryPointNode"))
             {
-                var entryPointNode = new EntryPointNode(env, ch, catalog, context, context.GenerateId(entryPointName), entryPointName,
+                var entryPointNode = new EntryPointNode(env, ch, context, context.GenerateId(entryPointName), entryPointName,
                     inputBuilder.GetJsonObject(arguments, inputBindingMap, inputMap),
-                    outputHelper.GetJsonObject(outputMap), checkpoint, stageId, cost);
+                    outputHelper.GetJsonObject(outputMap), checkpoint, stageId: stageId, cost: cost);
```
Comment (eerhardt) [inline src/Microsoft.ML.Data/EntryPoints/EntryPointNode.cs]: Is this change necessary?

File: src/Microsoft.ML.Data/EntryPoints/EntryPointNode.cs [hunk 3]
```diff
@@ -935,7 +929,7 @@ private object BuildParameterValue(List<ParameterBinding> bindings)
                         ch.Warning("Node '{0}' has unexpected fields that are ignored: {1}", id, string.Join(", ", unexpectedFields.Select(x => x.Name)));
                     }
 
-                    result.Add(new EntryPointNode(env, ch, catalog, context, id, nodeName, inputs, outputs, checkpoint, stageId, cost, label, group, weight, name));
+                    result.Add(new EntryPointNode(env, ch, context, id, nodeName, inputs, outputs, checkpoint, stageId: stageId, cost: cost, label: label, group: group, weight: weight, name: name));
```
Comment (eerhardt) [inline src/Microsoft.ML.Data/EntryPoints/EntryPointNode.cs]: Are the named parameters at the end of this line necessary?  `stageId: stageId, cost: cost, label: label, group: group, weight: weight, name: name));`

File: src/Microsoft.ML.Legacy/Runtime/EntryPoints/TrainTestMacro.cs [hunk 4]
```diff
@@ -121,10 +121,10 @@ public sealed class Output
 
             // Parse the subgraph.
             var subGraphRunContext = new RunContext(env);
-            var subGraphNodes = EntryPointNode.ValidateNodes(env, subGraphRunContext, input.Nodes, node.Catalog, input.LabelColumn,
-                input.GroupColumn.IsExplicit ? input.GroupColumn.Value : null,
-                input.WeightColumn.IsExplicit ? input.WeightColumn.Value : null,
-                input.NameColumn.IsExplicit ? input.NameColumn.Value : null);
+            var subGraphNodes = EntryPointNode.ValidateNodes(env, subGraphRunContext, input.Nodes, label: input.LabelColumn,
+                group: input.GroupColumn.IsExplicit ? input.GroupColumn.Value : null,
```
Comment (eerhardt) [inline src/Microsoft.ML.Legacy/Runtime/EntryPoints/TrainTestMacro.cs]: Do we need these named parameters?

File: src/Microsoft.ML.Data/EntryPoints/EntryPointNode.cs (hunk 1 above)
Comment (beneyal) [reply src/Microsoft.ML.Data/EntryPoints/EntryPointNode.cs]: InputBuilder's constructor takes IExceptionContext as first argument, not IHostEnvironment, and IExceptionContext doesn't have a ComponentCatalog property, so I think this one can't be changed.

**Comment by beneyal** [comment]
The named parameters thing is probably an artefact of ReSharper, I removed them.

File: src/Microsoft.ML.Data/EntryPoints/EntryPointNode.cs (hunk 1 above)
Comment (eerhardt) [reply src/Microsoft.ML.Data/EntryPoints/EntryPointNode.cs]: Ah, good point. Please ignore.
